### ISSUE TITLE
[$1000] Settings - There is a lag when transitioning between Account settings

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#47716]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.22-5
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Issue reported by:** Applause - Internal Team

## Action Performed:
1. Navigate to https://staging.new.expensify.com/
2. Log in with a new Gmail account
3. Navigate to Account settings - About
4. Click on "Inbox"
5. Click on "Account settings"

## Expected Result:
I should be able to navigate smoothly

## Actual Result:
The app freezes after opening Account settings again on specific pages. Affects most of the pages on Account settings. The freeze can last from 2-5 seconds to a point where you have to restart the browser. Sometimes the app returns to normal after you click

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/user-attachments/assets/3dfa8c19-c05b-4901-a1a9-df0d208acffc

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021846341339697727091</li>
        <li>Upwork Job ID: 1846341339697727091</li>
        <li>Last Price Increase: 2024-11-06</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @s77rt</details>

**Comment by lanitochka17** [comment]
@jliexpensify  FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by codewaseem** [comment]
I think same as https://github.com/Expensify/App/issues/47041

**Comment by jliexpensify** [comment]
Thanks @codewaseem - I think the root cause might be the same, as these issues are similar. I'll put a hold on this one, and then get Applause to re-test once https://github.com/Expensify/App/issues/47041 is resolved.

**Comment by jliexpensify** [comment]
Waiting on other PR

**Comment by jliexpensify** [comment]
@lanitochka17 can you re-test please? #47041 has been deployed to production and I can't reproduce this issue anymore,

**Comment by isagoico** [comment]
I'm still able to reproduce the issue on my side on 9.0.45-2. There's a notable freeze when navigating to About. 

https://github.com/user-attachments/assets/52f6e9fd-0e2a-4fc0-89e3-000d8da5d664

**Comment by adhorodyski** [comment]
@isagoico I'm able to only get 1/10 of this hang on my end, though performance profiling shows ~500ms is spent on setting up Lottie. I can also get rid of this hang completely when removing the animated sections. 

Could you please record a Chrome profiling session for us? Steps are here:
- right click somewhere on the page > Inspect
- navigate to the Performance tab
- follow https://developer.chrome.com/docs/devtools/performance#record to record a trace
- click the 'Save profile' on the left to the Record button & share it here

**Comment by isagoico** [comment]
@adhorodyski Tried again today and the freeze is not as notable as when I reproduced back [here](https://github.com/Expensify/App/issues/47716#issuecomment-2396890765). 

I recorded the Profile here [Profile.zip](https://github.com/user-attachments/files/17298073/Profile.zip).

**Comment by adhorodyski** [comment]
![Screenshot 2024-10-09 at 13 45 44](https://github.com/user-attachments/assets/0918334f-47b7-42e2-9ffa-ab930e387f5e)


Yup, looks like this is the same thing on your end.

**Comment by adhorodyski** [comment]
I've found [this PR](https://github.com/callstack-internal/Expensify-App/commit/3a3dafa958b1ebd6169ecae729f47fdecbb5c3e9) that postpones setting the asset so it delays the initiation - we can see the same in the Hermes trace as well. Looks like this was desired so offload & now we're just experiencing it each time we load this page?

```ts
useEffect(() => {
        setAnimationFile(source.file);
    }, [setAnimationFile, source.file]);
```

This block runs each time we mount this component to the screen so it might be that we just re-run the whole internal pipeline for Lottie.

**Comment by adhorodyski** [comment]
@jliexpensify the issue we're waiting for is closed now, what action can we take in this one?

**Comment by jliexpensify** [comment]
Hi @adhorodyski - hmm so I guess if this issue is still reproducible, we should dig into this further and try and fix the root cause. So we'd need to treat this as a separate issue to https://github.com/Expensify/App/issues/47041.

The other consideration is how bad this issue is, and whether it can actually be widely and consistently reproduced. If it's a niche issue, then maybe we can consider not fixing it? But we should probably bring this up publicly before making any decisions around it.

**Comment by adhorodyski** [comment]
It can be consistently reproduced, though I'd say it's not worth spending time on it at the moment unless we get more reports of this actually being problematic. I'm experiencing a 115-200ms (mac m1 pro) lag now due to this animation which is not that big of a deal (?) 

cc @mountiny curious about your opinion on this? I only want to prevent us chasing a 100ms lag on a settings page, but can dive deeper into this if we consider this being important (though we intentionally offloaded this lag to the subpages before).

**Comment by mountiny** [comment]
Hmm tough one. I would say we can make this external for $1000 and see if anyone can fix it in a root. I agree it might not be worth it having an agency worker on this, if we are only reproducing the lag

**Comment by jliexpensify** [comment]
Posted [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1729039805831709)

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by s77rt** [comment]
Not overdue. Looking for proposals

**Comment by s77rt** [comment]
Same ^

**Comment by Anaslancer** [comment]
@s77rt Let me confirm this error? It seems like we should fix the loading duration of the animations?

**Comment by s77rt** [comment]
@Anaslancer More specifically we are looking to understand why the app becomes laggy then fix it.

**Comment by jliexpensify** [comment]
Not overdue, still no proposals

**Comment by s77rt** [comment]
Not overdue ^

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Second week)

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Third week)

**Comment by s77rt** [comment]
Still looking for proposals

**Comment by jliexpensify** [comment]
@mountiny - coming back to this issue: what are your thoughts now? Should we close it, given it's been 3 weeks and no proposals?
